afk-daily.sh

```diff
@@ -1,1610 +1,1850 @@
-    getColor 700 670
-    while [ "$RGB" == "82fdf5" ]; do
-        input tap 930 680
-        wait
-        getColor 700 670
-    done
-
-    # Collect daily Chests
-    input tap 330 430 # Chest 20
-    wait
-    input tap 580 600 # Collect
-    input tap 500 430 # Chest 40
-    wait
-    input tap 580 600 # Collect
-    input tap 660 430 # Chest 60
-    wait
-    input tap 580 600 # Collect
-    input tap 830 430 # Chest 80
-    wait
-    input tap 580 600 # Collect
-    input tap 990 430 # Chest 100
-    wait
-    input tap 580 600 # Collect
-    wait
+    inputTapSleep 960 250                       # Quests
+    collectQuestChests_quick
 
     # Weekly quests
-    input tap 650 1650 # Weeklies
-    wait
-
-    # Collect weekly Quests
-    getColor 700 670
-    while [ "$RGB" == "82fdf5" ]; do
-        input tap 930 680
-        wait
-        getColor 700 670
-    done
-
-    # Collect weekly Chests
-    input tap 330 430 # Chest 20
-    wait
-    input tap 580 600 # Collect
-    input tap 500 430 # Chest 40
-    wait
-    input tap 580 600 # Collect
-    input tap 660 430 # Chest 60
-    wait
-    input tap 580 600 # Collect
-    input tap 830 430 # Chest 80
-    wait
-    input tap 580 600 # Collect
-    input tap 990 430 # Chest 100
-    wait
-    input tap 580 600 # Collect
-    wait
+    inputTapSleep 650 1650                      # Weeklies
+    collectQuestChests_quick
 
     # Return
-    input tap 70 1650
-    sleep 1
+    inputTapSleep 70 1650 1                     # Return
     verifyRGB 20 1775 d49a61 "Collected daily and weekly quest chests." "Failed to collect daily and weekly quest chests."
 }
 
-# Collects mail
+# ##############################################################################
+# Function Name : collectQuestChests_quick
+# Descripton    : Collects quest chests
+# ##############################################################################
+collectQuestChests_quick() {
+    if [ $DEBUG -ge 4 ]; then echo "[DEBUG] collectQuestChests_quick" >&2; fi
+    # Collect Quests
+    while testColorOR 700 670 82fdf5; do
+        inputTapSleep 930 680
+    done
+
+    if testColorNAND 350 380 54332b; then
+        inputTapSleep 330 430                   # Chest 20
+        inputTapSleep 580 600 0                 # Collect
+    fi
+    if testColorNAND 510 380 543323; then
+        inputTapSleep 500 430                   # Chest 40
+        inputTapSleep 580 600 0                 # Collect
+    fi
+    if testColorNAND 670 380 54331b; then
+        inputTapSleep 660 430                   # Chest 60
+        inputTapSleep 580 600 0                 # Collect
+    fi
+    if testColorNAND 830 380 533323; then
+        inputTapSleep 830 430                   # Chest 80
+        inputTapSleep 580 600 0                 # Collect
+    fi
+    if testColorNAND 1000 380 543323; then
+        inputTapSleep 990 430                   # Chest 100
+        inputTapSleep 580 600                   # Collect
+    fi
```

Comment:
I'm not sure I want to check if chests are collectable... In my experience, various patches commonly change the location of icons, if just ever so slightly, which means we will most likely have to keep updating the RGB values for each chest. So I think the amount of headache this potentially saves by just hard-coding it and making the script attempt to always collect every chest is worth it.

Though I did just remember that we now have a `delta` value, which comes in handy very much here. Are you making use of it in this case?

Reply:
It's a not equal, so if they change the value we will have false negative (so it will tap when it's not useful, not a big problem).
I'll add a `-d $DEFAULT_DELTA` if it's happening.

Reply:
Got it, did not see the `NAND` there, my bad.